includes/Pages/PageSearch.php

```diff
@@ -27,112 +29,167 @@ protected function main()
     {
         $this->setHtmlTitle('Search');
 
+        $database = $this->getDatabase();
+        $currentUser = User::getCurrent($database);
+
+        $this->assign('canSearchByComment', $this->barrierTest('byComment', $currentUser));
+        $this->assign('canSearchByEmail', $this->barrierTest('byEmail', $currentUser));
+        $this->assign('canSearchByIp', $this->barrierTest('byIp', $currentUser));
+        $this->assign('canSearchByName', $this->barrierTest('byName', $currentUser));
+        $this->assign('canSeeNonConfirmed', $this->barrierTest('allowNonConfirmed', $currentUser));
+
+        $this->setTemplate('search/main.tpl');
+
         // Dual-mode page
-        if (WebRequest::wasPosted()) {
-            $searchType = WebRequest::postString('type');
-            $searchTerm = WebRequest::postString('term');
+        if (WebRequest::getString('type') !== null) {
+            $searchType = WebRequest::getString('type');
+            $searchTerm = WebRequest::getString('term');
+
+            $excludeNonConfirmed = true;
+            if ($this->barrierTest('allowNonConfirmed', $currentUser)) {
+                $excludeNonConfirmed = WebRequest::getBoolean('excludeNonConfirmed');
+            }
 
             $validationError = "";
             if (!$this->validateSearchParameters($searchType, $searchTerm, $validationError)) {
                 SessionAlert::error($validationError, "Search error");
-                $this->redirect("search");
+
+                $this->assign('term', $searchTerm);
+                $this->assign('target', $searchType);
+                $this->assign('excludeNonConfirmed', $excludeNonConfirmed);
+                $this->assign('hasResultset', false);
```

Comment:
The `$limit` Smarty variable isn't assigned here, so no limit option is selected on validation failure. Was that intentional?